hawkular-alerts-metrics/src/test/groovy/org/hawkular/alerts/external/ExternalMetricsITest.groovy

```diff
@@ -620,6 +621,100 @@ class ExternalMetricsITest extends AbstractExternalITestBase {
         assertEquals("0.8", resp.data[0].evalSets[0].iterator().next().value)
     }
 
+    @Test
+    void t08_heartbeatTest() {
+        // use a unique metricId for the test run so that we don't conflict with prior test runs
+        String availId = "alerts-test-heartbeat-" + start;
+
+        // CREATE trigger using external metrics expression
+        def resp = client.get(path: "")
+        assertTrue(resp.status == 200 || resp.status == 204)
+
+        Trigger triggerTestHeartbeat = new Trigger("trigger-test-heartbeat", "trigger-test-heartbeat");
+
+        // remove if it exists
+        resp = client.delete(path: "triggers/trigger-test-heartbeat")
+        assertTrue(200 == resp.status || 404 == resp.status)
+
+        triggerTestHeartbeat.setAutoDisable(true);
+
+        // Tag the trigger as a HawkularMetrics:MetricsCondition so it gets picked up for processing
+        triggerTestHeartbeat.addTag("HawkularMetrics", "MetricsCondition" );
+
+        resp = client.post(path: "triggers", body: triggerTestHeartbeat)
+        assertEquals(200, resp.status)
+
+        // ADD external metrics heartbeat condition. Note: alerterId must be "HawkularMetrics"
+        // Average over the last 1 minute > 50, check every minute
+        ExternalCondition firingCond = new ExternalCondition("trigger-test-heartbeat", Mode.FIRING, "external-dataId-heartbeat",
+            "HawkularMetrics", "metric:1:heartbeat(" + availId + " <= 2),1");
```

Comment:
This is a good point.  The reason it works today is because the external alerter for metrics assumes GAUGE metrics (unless its an avail-related expression).  Because the external alerter is in charge of evaluating the expression, we can make changes as needed.  One possibile enhancement may be to specify the metric type as the expression "target" (see the doc for what that means).  But considering this stuff is not yet used, we have time to make changes as needed.